### DISCUSSION TITLE
How to set indivisual timeout in test Library

### DISCUSSION THREAD

**Discussion by nair-sumesh** [ui5-community/wdi5#407] [Q&A]
How to set the timeout at element level in Test Library with WDI5.     
For own test functions, to set the timeout is as follows
```js
await browser.asControl({
			timeout: 90000,
                        ...
)}
```

Currently my test for
```js
await oFioriElementsFacade.execute((Given, When, Then) => {
		Then.onTheDetailPage.iSeeThisPage();
		When.onTheDetailPage
			.onForm("General Information")
			.iChangeField("Name","Testing")
		Then.onTheDetailPage.iSeeThisPage()
	})
```
 fails with message
```cmd
Polling stopped because the timeout of 15000 milliseconds has been reached but there is still pending asynchronous work.
```
Regards,    
Sumesh Nair

**Comment by Siolto** [comment]
Hi Sumesh,

currently only the default of 15 seconds is available. But there is already an [branch](https://github.com/ui5-community/wdi5/compare/main...feat/fe_lib_timeout) available which increases the timeout of the FE Testlib depending on your `waitForUI5Timeout` property in the `conf.js` file.

Regards,
Simon

**Reply by nair-sumesh** [reply]
Thanks @Siolto for the prompt response.
Will this also allow to adjust the timeout at element / step level.

I would like to keep the global timeout to 15 seconds and increase the timeouts for specific steps.

**Reply by Siolto** [reply]
As far as I am familiar with the fiori elements test library it is not possible to set different timeouts for specific steps. I found no `timeout` property in the [API](https://sapui5.hana.ondemand.com/#/api/sap.fe.test) which you could use.

As we are just using the test library in `wdi5` this would be something the FE Team has to develop.

Best regards,
Simon

**Reply by nair-sumesh** [reply]
Hi @Siolto ,

Is it possible to introduce a function which can simply overwrite the `opaConfig: { timeout: 90 }` for the current ( or remaining of the ) test.    

Regards,    
Sumesh
